### DISCUSSION TITLE
优化检测OpenResty应用的接口

### DISCUSSION THREAD

**Discussion by jamebal** [1Panel-dev/1Panel#4886] [Q&A]
### 1Panel 版本

v1.10.5-Its

### 请描述您的需求或者改进建议

1. 打开网站页面和WAF界面会很慢, 非常影响使用体验, 主要是这两个接口, 响应在500ms以上, 我以为是我安装的docker应用太多了, 结果用一个新的系统安装1Panel, 这两个接口还是很慢
<img width="1126" alt="截屏2024-04-25 08 59 51" src="https://github.com/1Panel-dev/1Panel/assets/49510841/263af345-c613-4e22-bc9b-1ad7ea2fabff">
<img width="1090" alt="截屏2024-04-25 09 00 17" src="https://github.com/1Panel-dev/1Panel/assets/49510841/94779ba9-0ad6-4e46-a341-cfe654813102">
2. 打开WAF界面时会调用两次`/api/v1/waf/status`接口

**Comment by jamebal** [comment]
还有个接口也很慢: 
<img width="1082" alt="截屏2024-04-25 09 17 26" src="https://github.com/1Panel-dev/1Panel/assets/49510841/6b012ec7-b1c7-432c-bfdd-75dbaaab88e0">

**Comment by zhengkunwang223** [comment]
请求速度可能与你的机器网络和性能有关
你是哪里的机器 配置如何

**Comment by jamebal** [comment]
网络是局域网
机器是自己搭的服务器, 64H 128G

**Comment by jamebal** [comment]
从昨天晚上到今天早上都很慢, 现在又好了不知道是什么问题
请问这些接口里面有通过网络请求其他数据吗

**Comment by jamebal** [comment]
<img width="1110" alt="截屏2024-04-25 13 59 15" src="https://github.com/1Panel-dev/1Panel/assets/49510841/4e83a073-7eb0-457e-ba10-8c675c8a09bf">
又出现了, 不知道什么问题, 网络和机器性能没有问题,而且机器负载很低的

**Comment by jamebal** [comment]
经过多次测试, 可能是WAF的问题, 把自己的IP加入白名单, 以上接口正常, 移出白名单就很慢

**Comment by zhengkunwang223** [comment]
![image](https://github.com/1Panel-dev/1Panel/assets/31820853/ec45ce26-0cf3-4387-b927-53565b376eff)


这是我的境外 1Panel 开启 WAF 并使用没有优化的 cloudflare 的打开速度  不知道你的是什么情况 
看你是自己搭的服务器 是不是内网穿透没配置好呢

**Comment by jamebal** [comment]
我在其他服务器上也安装1panel测试了一下, 没出现这个问题, 只有在这个服务器上有这个问题;
这个服务器配置: CPU: AMD 7d12 64核, 内存128G;
关掉WAF问题依然存在, 索引也不是WAF的问题, 可能是我服务器的问题, 但是为什么呢?

1. 性能问题: 在服务器负载差不多的情况下, 这几个接口有时正常,有时很慢, 所以不是性能问题;
2. 网络问题: 局域网环境, 不访问外网, 所以不是网络问题;
3. 服务器上的其他应该是否会影响到这个几个接口的响应速度?
4. 这几个接口有时正常,有时很慢,目前没发现规律, 因为几个接口在响应慢的时候, 其他1panel的接口都是正常的, 而且服务器的其他应用也都正常

麻烦大佬帮我分析一下, 谢谢🙏

在虚拟机里安装的1panel, 目前有53个docker容器
<img width="1448" alt="截屏2024-04-27 11 34 23" src="https://github.com/1Panel-dev/1Panel/assets/49510841/65f7fdec-0e53-49fa-99bd-543a9f0fa01d">

**Comment by jamebal** [comment]
在多个虚拟机里面安装1panel测试发现: 
1. 系统里docker容器越多(多于30个),`/v1/apps/installed/check`, `/v1/apps/installed/search`这两个接口就慢, 不总是很慢, 有时几百毫秒,有时几秒, 慢点程度基本和docker容器数量成正比
2. 系统里docker容器少(少于10个), `/v1/apps/installed/check`, `/v1/apps/installed/search`这两个接口就很快, 总是很快, 200毫秒以内
3. 以上测试来自同一物理机里的多个虚拟机测试, 虚拟机配置相同, 所以能排除网络和硬件配置性能的问题

**Reply by zhengkunwang223** [reply]
感谢反馈 我们测试一下
